purchase_allowed_product/models/supplied_product_mixin.py

```diff
@@ -8,15 +8,27 @@ class SuppliedProductMixin(models.AbstractModel):
     _name = "supplied.product.mixin"
     _description = "Mixin for documents that want to restrict products"
 
+    def _default_hide_use_only_supplied_product(self):
+        return self.user_has_groups(
+            "purchase_allowed_product.group_purchase_hide_allowed_product_option"
+        )
+
     use_only_supplied_product = fields.Boolean(
         string="Use only allowed products",
         help="If checked, only the products provided by this supplier "
         "will be shown.",
     )
 
+    hide_use_only_supplied_product = fields.Boolean(
+        'Hide "Use only allowed products" in PO and invoice',
+        default=_default_hide_use_only_supplied_product,
+        readonly=True,
+    )
+
     @api.onchange("partner_id")
     def _onchange_partner_id_supplied_product(self):
         self.use_only_supplied_product = (
-            self.partner_id.use_only_supplied_product
+            self.use_only_supplied_product
```

Comment:
With this, you are avoiding the option to disable the check automatically if previously was set. This is an undesired behavior.